Bazel: introduce a message flag to run bazel jobs (#49169)
sourcegraph/sourcegraph: doc/dev/background-information/bazel.md

Sourcegraph is currently migrating to Bazel as its build system and this page is targeted for early adopters which are helping the [#job-fair-bazel](https://sourcegraph.slack.com/archives/C03LUEB7TJS) team to test their work.

## Early adopters 

If you are an early adopter, you can already get some benefits from Bazel, while we gradually roll it out.  

> :bulb: Please not that this is only applicable in PR, as for the `main` branch we need to ensure we same the same build steps for everyone until Bazel is fully rolled out. 

Reviewer: ```suggestion
:bulb: Please note that this is only applicable in PRs. For the `main` branch we need to ensure we use the same build steps for everyone until Bazel is fully rolled out. 
```

- Before pushing, ensure your changes are refected in the build files (those `BUILD.bazel` files):
  - If you changed anything to the `go.mod` file, you need to run: 
    - `bazel run :update-gazelle-repos` 
  - Run `bazel configure` to ensure the build files are also properly updated.
- Run your tests locally, with `bazel test //[PATH]/...` where `PATH` refers to the package containing your changes. 
  - If you changed things in too many places, you can always run `bazel test //...` which will test everything (or reused cached results if applicable). 
- Include the updated build files in your commit! They are relevant to that commit after all. 

> - When committing, add the `[force-bazel]` message flag in the description of your commit (not on the commit title, but in the description, it's nicer this way).

Reviewer: ```suggestion
- When commiting, add the `[force-bazel]` message flag in the description of your commit (not in the commit title, but in the description - it's nicer this way).
```

>   - If you commit again, remember to add that message flag again, only the last commit is checked for to determine if we want Bazel on that PR. 

Reviewer: ```suggestion
  - If you commit again, remember to add that message flag again. Only the last commit is checked to determine if we want Bazel on that PR. 
```

- Push your changes as usual. 
- When browsing the CI (you can use `sg ci status --web` you'll see a `Bazel` set of jobs running both your tests and build. 

You may find the build and tests to be slow, either locally or in CI. This is because to be efficient, Bazel cache needs to be warm. So inevitably, as early adopters, that will be less the case 
than when more teammates will be using Bazel. 

> :warning: It's highly probable that your build files updates will include changes that you were not responsible for. This is because not everyone is updating buildfiles. Just commit them anyway and move on. This will be get better over time. 

Reviewer: ```suggestion
:warning: It's highly probable that the build files you updated will include changes that you were not responsible for. This is because not everyone is updating buildfiles. Just commit them anyway and move on. This will be get better over time. 
```
